tests/Flarum/Admin/Middleware/LoginWithCookieAndCheckAdminTest.php

```diff
@@ -0,0 +1,27 @@
+<?php
+namespace tests\Flarum\Admin\Middleware;
+
+use Flarum\Admin\Middleware\LoginWithCookieAndCheckAdmin;
+use Flarum\Core\Exceptions\PermissionDeniedException;
+use Illuminate\Contracts\Container\Container;
+use Mockery as m;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use tests\Test\TestCase;
+
+class LoginWithCookieAndCheckAdminTest extends TestCase
+{
+    public function test_it_should_not_allow_invalid_logins()
+    {
+        $this->setExpectedException(PermissionDeniedException::class);
+
+        $container = m::mock(Container::class);
+        $request = m::mock(ServerRequestInterface::class);
+        $response = m::mock(ResponseInterface::class);
+
+        $request->shouldReceive('getCookieParams')->andReturn([]);
```

Comment:
Yeah I could set it up as a spy, but the reason I've done it this way is to pave the way for tests once I can test the class in full. Currently there is too much embedded code that is untestable.